Doc/library/tempfile.rst

```diff
@@ -75,20 +75,52 @@ The module defines the following user-callable items:
       Added *errors* parameter.
 
 
-.. function:: NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, delete=True, *, errors=None)
-
-   This function operates exactly as :func:`TemporaryFile` does, except that
-   the file is guaranteed to have a visible name in the file system (on
-   Unix, the directory entry is not unlinked).  That name can be retrieved
-   from the :attr:`name` attribute of the returned
-   file-like object.  Whether the name can be
-   used to open the file a second time, while the named temporary file is
-   still open, varies across platforms (it can be so used on Unix; it cannot
-   on Windows).  If *delete* is true (the default), the file is
-   deleted as soon as it is closed.
-   The returned object is always a file-like object whose :attr:`!file`
-   attribute is the underlying true file object. This file-like object can
-   be used in a :keyword:`with` statement, just like a normal file.
+.. function:: NamedTemporaryFile(mode='w+b', buffering=-1, encoding=None, newline=None, suffix=None, prefix=None, dir=None, delete=True, *, errors=None, delete_on_close=True)
+
+   This function operates exactly as :func:`TemporaryFile` does, except the
+   following differences:
+
+   * The file is guaranteed to have a visible name in the file system
+     (on Unix, the directory entry is not unlinked).
```

Comment:
I know this line was here before, but it doesn't make any sense to me. Should we mention this somewhere else?

Reply:
@eryksun , I think zooba talks about the line "(on Unix, the directory entry is not unlinked)." What do you think, does it make scene to move it somewhere else?

Reply:
On POSIX, a file created by `TemoraryFile()` does not have a visible name in the filesystem. The implementation tries to use the `O_TMPFILE` flag if it's supported (Linux 3.11+), which creates an anonymous file. If `O_TMPFILE` isn't supported, the file is created with a unique name, and then it's immediately unlinked while the file descriptor is still open.

On Windows, `TemporaryFile` is aliased to `NamedTemporaryFile`. There isn't a compelling reason for this decision. It's fine to delete a file that's opened with the `O_TEMPORARY` flag. On Windows 10 and 11, if it's on an NTFS filesystem, the file gets unlinked immediately. Prior to Windows 10, or on filesystems other than NTFS, the file is deleted but not unlinked. In the latter state, a file has a visible entry in the directory, but it can't be opened again for any access, and it gets unlinked as soon as the last handle to it is closed.

Reply:
@eryksun , thanks for explanation. So, would you agree, that the statement _on Unix, the directory entry is not unlinked_ does not really make too much scene the way it is written now? Would it be better to either remove it completely or to put a broader explanation, similar to what you provided?

Reply:
It's in the context of clarifying that `NamedTemporaryFile()` acts "exactly" like `TemporaryFile()`. In that context it makes sense. But I'd prefer that the first sentence and the two bullet points were replaced by something like the following:

```text
   This function returns a file that is guaranteed to have a visible name in
   the file system. To manage the named file, it extends the parameters of
   :func:`TemporaryFile` with *delete* and *delete_on_close* parameters that
   control whether and how the named file should be automatically deleted.
```

Reply:
I have now placed the statement about unlinking on Unix to a different place and slightly reworked it to make a reference to TemporaryFile(), without which I agree it was a bit confusing.

I have also implemented rewording from @eryksun